includes/Modules/Analytics.php

```diff
@@ -1232,4 +1239,76 @@ private function is_adsense_request( $data ) {
 	protected function setup_settings() {
 		return new Settings( $this->options );
 	}
+
+	/**
+	 * Checks if the module is active in the admin bar for the given URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $url URL to determine active state for.
+	 * @return bool
+	 */
+	public function is_active_in_admin_bar( $url ) {
+		if ( ! $this->is_connected() ) {
+			return false;
+		}
+
+		return $this->has_data_for_url( $url );
+	}
+
+	/**
+	 * Checks whether Analytics data exists for the given URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $url The url to check data for.
+	 * @return bool
+	 */
+	protected function has_data_for_url( $url ) {
+		if ( ! $url ) {
+			return false;
+		}
+
+		$transient_key = 'googlesitekit_analytics_has_data_' . md5( $url );
+		$has_data      = get_transient( $transient_key );
+
+		if ( false === $has_data ) {
+			/* @var Google_Service_AnalyticsReporting_Report[]|WP_Error $reports Array of reporting report instances. */
+			$reports = $this->get_data(
+				'report',
+				array(
+					'url'     => $url,
+					'metrics' => array(
+						array( 'expression' => 'ga:users' ),
+						array( 'expression' => 'ga:sessions' ),
+					),
+				)
+			);
+
+			if ( is_wp_error( $reports ) ) {
+				return false;
+			}
```

Comment:
We should not return `false` here, but instead cache `0` for one hour as well, as if no data had been found. Otherwise there is a risk of querying this endpoint on every pageload.